ddtrace/appsec/processor.py

```diff
@@ -210,9 +217,51 @@ def __attrs_post_init__(self):
         # we always need the response headers
         self._mark_needed(_Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES)
 
+    def update_rules(self, new_rules):
+        # type: (List[Dict[str, Any]]) -> None
+        try:
+            self._ddwaf.update_rules(new_rules)
+        except TypeError as e:
+            log.debug("Error updating RCM rules: %s", str(e))  # noqa: G200
+
     def on_span_start(self, span):
         # type: (Span) -> None
-        pass
+
+        peer_ip = _asm_context.get_ip()
+        headers = _asm_context.get_headers()
+        headers_case_sensitive = _asm_context.get_headers_case_sensitive()
+
+        if headers is not None:
+            _context.set_items(
+                {
+                    "http.request.headers": headers,
+                    "http.request.headers_case_sensitive": headers_case_sensitive,
+                },
+                span=span,
+            )
+
+            if peer_ip is not None:
+                ip = trace_utils._get_request_header_client_ip(headers, peer_ip, headers_case_sensitive)
+                # Save the IP and headers in the context so the retrieval can be skipped later
+                _context.set_item("http.request.remote_ip", ip, span=span)
+                if ip and self._is_needed(_Addresses.HTTP_CLIENT_IP):
+                    data = {_Addresses.HTTP_CLIENT_IP: ip}
+                    log.debug("[DDAS-001-00] Executing AppSec In-App WAF with parameters: %s", data)
```

Comment:
```suggestion
                    log.debug("[DDAS-001-00] Executing ASM WAF with parameters: %s", data)
```

this will log IPs in plaintext is this a data sensitivity concern?

Reply:
Maybe. I will remove it just in case. Thanks!